README.md

```diff
@@ -24,6 +24,57 @@ In order to limit cognitive overhead prefer:
 MVC + `actions` + `jobs` + `services` + `workers`
 
 - Actions receive & validate user input that create/update records, with any related business logic.
-- Job have a single purpose with a clear start and end, usually asynchronous and ideally idempotent. 
+- Job have a single purpose with a clear start and end, usually asynchronous and ideally idempotent.
 - Services encapsulate complex logic or external calls that can be used in multiple places, usually jobs and actions.
 - Workers are long running processes listening for events, usually triggering relevant job in response.
+
+
+## Front end structure, rules and guidelines
+To keep the code clean, consistent and well understood across all projects, we should follow the conventions, rules and guidelines below, in addition to the rules and conventions specified by the linting rules and generally those adopted by the language or framework/s.
+
+General conventions
+- directories should use camelCase multiword names
+- regular js modules should follow the same directory naming conventions for components, and should be named in camelCase
+- if the js module exports a class, the filename should be in PascalCase
+- avoid relative imports -- prefer absolute paths always. http://codeofrob.com/entries/stop-using-relative-paths-in-your-javascripts.html
+
+Component Naming Conventions:
+- always use multiword component names to avoid naming collisions or nondescriptive components
+ - you should always know exactly what it is referring to based on the name
+- use PascalCase to name all vue components
+ - e.g. `TodoItem`
+- child components that are tightly coupled with the parent should include the parent component in its name as a prefix - https://vuejs.org/v2/style-guide/#Tightly-coupled-component-names-strongly-recommended
+- component names should start with the highest (most general) words and end with the descriptive words - https://vuejs.org/v2/style-guide/#Order-of-words-in-component-names-strongly-recommended
+- component names should always prefer full words over abbreviations - https://vuejs.org/v2/style-guide/#Full-word-component-names-strongly-recommended
+- avoid nested levels of folders
+ - the components should be grouped well and easily discoverable by name, and forces developers to give descriptive names following the naming conventions
+
+Page Naming Conventions:
+- page components should be prefixed with the resource they refer to
+  - `OrganisationsTable`, `BriefGroupForm`
+- for pages that list content for a resource, suffix pages with `List`
+  - `OrganisationsList`, `BriefGroupsList`
+- we should use the same page component when building create/edit pages, and the resource should be singular and suffixed with `View`
+  - `OrganisationView`, `BriefGroupView`
```

Comment:
Would we want to distinguish at all between "show", "new", "edit" components or can they usually all be `XxxView`?

Reply:
In most cases (haven't seen one yet that isn't), the new/edit components would be identical -- we display a form or something to input/edit the data and save it to the database. To avoid duplication, it would be preferable to choose one naming convention for those pages instead of having two separate pages.

I suggested the above because it is rare where we only show the data for a single resource by itself that isn't tied to some sort of form. So we can either choose to use `XxxView` for those pages, or use something specific for the create/edit pages such as `XxxEdit`.

Reply:
I would suggested `XxxEdit` for write components, `XxxView` for read components, just so it's clear just looking at the name what they can do.

Reply:
Yeah that sounds pretty reasonable to me

Reply:
Let's make the rule that `Edit` CAN be used for both `Create` and `Edit` but that `Create` should be created next to `Edit` if the functionality calls for it or if the subject cannot be updated after it's initial creation.

Reply:
I feel it could get confusing if we have create and edit pages because a lot of the time when building a new page, people often copy/paste or follow how a similar page was built and unless it is clear they might unnecessarily both. If we can all get on the same page for this though, I think that's fair in certain circumstances, but maybe not anything we have currently. Happy to vote on this one

Reply:
Let's keep it `Edit` for now. Special circumstances allow for special naming IMHO.